Compatibility v13

WebMaestroFr opened the pull request:
:warning: I merged https://github.com/ForgeVTT/forge-compendium-browser/pull/41 to `main` by mistake when creating this branch. The only thing I managed to revert was the git history somehow, but the changes are still on there.

WebMaestroFr committed:
Set globals for ESLint, clean and fix errors

WebMaestroFr committed:
Fix build command

WebMaestroFr committed:
Fix warnings

WebMaestroFr committed:
Merge branches 'main' and '2024-06-08-eslint-prettier' of github.com:ForgeVTT/forge-compendium-browser

WebMaestroFr committed:
Merge branch 'v12-compatibility' of github.com:ForgeVTT/forge-compendium-browser

WebMaestroFr committed:
Fix CSS syntax error

WebMaestroFr committed:
WIP Use ApplicationV2

WebMaestroFr committed:
Revert formatting changes on Vue components

WebMaestroFr committed:
Improve retro-compatibility with <v12

WebMaestroFr committed:
Clean up

WebMaestroFr committed:
Fix Compendium browser

zeel01 commented on src/forge-compendium-browser.js:
```diff
@@ -160,6 +160,7 @@ export class ForgeCompendiumBrowser {
     }
 
     static async onMessage(data) {
+        console.warn("ForgeCompendiumBrowser.openBrowser", data);
```
Warning?

zeel01 reviewed (changes requested):
Looks good for the most part, but I think we need to clean up the console output. If something is really important for debugging we should prefer `console.debug()` as the verbose log level if nothing is going wrong. I was going to suggest we could use the log levels from https://foundryvtt.com/packages/_dev-mode but it seems to not have been updated past v9...

Eranziel reviewed (changes requested):
A good start! Still needs some work, could be done in a follow-up PR.
- Light mode CSS is borked.
  <img width="1735" height="769" alt="Image" src="https://github.com/user-attachments/assets/1d1cea5e-1306-4afa-98ad-6b51692d33fa" />
  <img width="1735" height="769" alt="Image" src="https://github.com/user-attachments/assets/1e2e3fe9-c8a4-444d-9d09-723607532708" />
- Actor sheets don't render in a usable way.
  <img width="1365" height="724" alt="Image" src="https://github.com/user-attachments/assets/3f396eb8-282e-4a64-b07b-fa6395e22236" />
- Item sheets don't render in a usable way.
  <img width="1365" height="724" alt="Image" src="https://github.com/user-attachments/assets/505178da-5203-45ab-8b1c-0522464fcf2e" />
- Journals don't render. (Vanilla journal show behind compendium browser window.)
  <img width="1766" height="864" alt="Image" src="https://github.com/user-attachments/assets/0d703027-d689-45f9-9585-649636397203" />

zeel01 committed:
Updated the code to hide compendium entries in the sidebar

zeel01 committed:
Removed older version of base app

zeel01 committed:
Added support for dark mode

zeel01 committed:
Refactored inner document rendering flow.
- Improved code organization
- Made rendering compatible with v13 applications
- Made rendering compatible with new 5e system

zeel01 committed:
Added development section to Readme

zeel01 committed:
Removed redundant classes from wrapper, classes are needed on the "frame" element instead.

zeel01 committed:
Adjusted general alignments and styles

zeel01 committed:
Updated the way sub-document sheets are displayed within the app.

zeel01 committed:
Made listing group background contrast better in dark mode

zeel01 committed:
Removed some more jquery from the scene section

zeel01 committed:
Fixed imports

zeel01 committed:
Fixed item tabs

zeel01 committed:
Made breadcrumb bar text size consistent with previous version

zeel01 committed:
Tweaked the display of sub-document sheets

zeel01 committed:
Replaced some warnings with debug output instead

zeel01 committed:
Updated package manifest

zeel01 commented:
- Substantially refactored some portions of the code for entries. Partially because it needed cleaned up, but mainly because the process of refactoring it let me learn how it worked and update it for v13.
- Light and dark mode are now both supported
- Actor and Item sheets render correctly inside the app
- Journal sheets render correctly inside the app
- Imports now work
- DNDB compendiums are correctly hidden from the sidebar
- Added a little advice to the readme

zeel01 commented:
@qalucaspacheco1 QA notes:
- Install dependencies: `dlopen`, `vueport`
- Install package via import wizard: [forge-compendium-browser.zip](https://github.com/user-attachments/files/25407497/forge-compendium-browser.zip)
- The previous verified compatibility for this module was v11, so we primarily want to know if there are any regressions between the published version in v11 and this new version in v13. 
- The v11 version *does* work in v12 but has a few issues. This v13 version is not compatible with older versions of Foundry. I have not tested it on v14.
- Neither of the dndb basic rules packs contain any scenes, you will need to import a dndb module that has scenes to test that part. I'm not sure how you can do that on QA unless you own a book with maps.

zeel01 committed:
Modified readme

Eranziel reviewed (changes requested):
A few comments and nit-picks, shouldn't be anything major. The main thing remaining is putting this through QA.

zeel01 committed:
Update debug message to match method name

zeel01 committed:
Ignore zip files

zeel01 committed:
Revert iterating version number a second time

zeel01 committed:
Cleaned up commented out code

zeel01 committed:
Removed unneeded version check

zeel01 committed:
Removed unnecessary base class.

zeel01 committed:
Force the sheet to not be editable.

Eranziel reviewed (approved):
LGTM!
